_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -5079,31 +5079,111 @@ mfxU64 make_back_color_yuv(mfxU16 bit_depth, mfxU16 Y, mfxU16 U, mfxU16 V)
     VM_ASSERT(bit_depth);
 
     mfxU64 const shift = bit_depth - 8;
-    mfxU64 const max_val = 256 << shift;
+    mfxU64 const max_val = 1 << bit_depth;
 
     return
         ((mfxU64)                                 (max_val - 1) << 48) |
         ((mfxU64)VPP_RANGE_CLIP(Y, (16 << shift), (235 << shift))     << 32) |
         ((mfxU64)VPP_RANGE_CLIP(U, (16 << shift), (240 << shift))     << 16) |
         ((mfxU64)VPP_RANGE_CLIP(V, (16 << shift), (240 << shift))     <<  0)
         ;
-};
+}
+
+inline
+mfxU64 make_back_color_argb(mfxU16 bit_depth, mfxU16 R, mfxU16 G, mfxU16 B)
+{
+    VM_ASSERT(bit_depth >=8);
+
+    mfxU64 const max_val = 1 << bit_depth;
+
+    return ((mfxU64) (max_val - 1) << 48) |
+           ((mfxU64) VPP_RANGE_CLIP(R, 0, max_val - 1) << 32) |
+           ((mfxU64) VPP_RANGE_CLIP(G, 0, max_val - 1) << 16) |
+           ((mfxU64) VPP_RANGE_CLIP(B, 0, max_val - 1) << 0);
+}
 
 inline
 mfxU64 make_def_back_color_yuv(mfxU16 bit_depth)
 {
-    assert(bit_depth);
+    // A = max, Y=min, U,V = max/2
+    VM_ASSERT(bit_depth >= 8);
 
     mfxU64 const shift = bit_depth - 8;
-    mfxU64 const min_val = 16 << shift, max_val = 256 << shift;
+    mfxU16 const min_val = 16 << shift, max_val = 256 << shift;
+
     return
-        ((max_val - 1ULL) << 48) |
-        ( min_val      << 32) |
-        ( max_val / 2  << 16) |
-        ( max_val / 2  <<  0)
-        ;
+        ((mfxU64)(max_val - 1) << 48) |
+        ((mfxU64)min_val       << 32) |
+        ((mfxU64)max_val / 2   << 16) |
+        ((mfxU64)max_val / 2   <<  0);
+}
+
+inline
+mfxU64 make_def_back_color_argb(mfxU16 bit_depth)
+{
+    // A = max, R,G,B = 0
+    VM_ASSERT(bit_depth >= 8);
+
+    mfxU64 const max_val = 1 << bit_depth;
+
+    return ((mfxU64)(max_val - 1) << 48);
 };
 
+static
+mfxU64 get_background_color(const mfxVideoParam & videoParam)
+{
+    mfxU64 back_color = make_def_back_color_argb(8);
```

Comment:
Bad idea. With this code we will end up returning rgb back color for non-rgb formats. I think better to move this in the switch and assign honest 0 + add assert in the "default" switch case. Well, you also can change function prototype returning mfxStatus.